### PR TITLE
Upgrade RoundCube to 1.4.1 and set the default skin to elastic

### PR DESCRIPTION
**Pull request by fmonts**
RoundCube 1.4 has been released.
The most notable feature is a new skin, responsive and with a much more modern look.

It has been in release candidate for 9 months, with hundreds of commits, so I think we can consider it stable and set it by default.

### CODE REVIEW

**Comment by ctrl-i** [comment]
I have already submitted a PR for this update [here](https://github.com/mail-in-a-box/mailinabox/pull/1671)

I did not add the part to upgrade the default theme to elastic in case this broke some users experience.  I left it up to the user to change the the new theme.

**Comment by fmonts** [comment]
Yes but most users have no way to know that a new skin has been released, and they'll never change it. 
In this way, if they don't like it, they can still go back to the settings and set the old one. 

From my experience, most of the users are accessing the webmail from mobile devices, and I had many complaints about the fact that it's not responsive.

Also, for the newly created users, I don't see a reason to create them with the old skin (unless the new one is bugged, I've been trying it and I found no bugs so far)

**Comment by ctrl-i** [comment]
It is still a duplicate PR :-)

**Comment by fmonts** [comment]
Yeah sorry, didn't see it... I also asked in the forum first https://discourse.mailinabox.email/t/update-to-roundcube-1-4/5655

**Comment by fmonts** [comment]
Lol there's no need to get upset if another PR is better :)) We don't get any money if a PR is merged, we're just serving the community ;)

**Comment by ctrl-i** [comment]
@fmonts Pathetic.  I couldn't care less which is better.  I would though have the decency to close a duplicate PR if I created one.  Sorry (not really) I beat you in creating this one.  I will close my PR so as not to clutter up MIAB and not bother posting PR in the future.  I hope you continue to serve the community with your quality pull requests.

**Comment by fmonts** [comment]
"you beat me"... I don't understand why you see it as a race
I think they are conflicting PRs, the maintainers should decide which one to merge, (it they think the new skin should be the default or not), I don't see an issue in having both open until they decide

**Comment by JoshData** [comment]
@ctrl-i @fmonts Thank you both. I prefer this one - our strategy is to always give all users the best experience without giving a choice.

@fmonts Would you add an entry in our CHANGELOG.md file?

Have either of you checked if Roundcube 1.4 has different dependencies that we should install (or any packages that are no longer needed)? The Roundcube upgrade notes also say "4. When upgrading from version older than 1.4.0 make sure old files in configured temp_dir are removed. Since this version we use constant filename prefix and do not remove files not starting with "RCMTEMP"." Do we need to do anything for that?

**Comment by ctrl-i** [comment]
@JoshData Ask @fmonts, I'm not interested in contributing any longer.

**Comment by fmonts** [comment]
> @fmonts Would you add an entry in our CHANGELOG.md file?

Done

> When upgrading from version older than 1.4.0 make sure old files in configured temp_dir are removed. Since this version we use constant filename prefix and do not remove files not starting with "RCMTEMP"." Do we need to do anything for that?

That means we should `rm /var/tmp/roundcubemail/RCMTEMP*`, correct? I don't have any files starting with RCMTEMP, both in the instance where I upgraded and in the others still at roundcube 1.3... Should we delete only those or all the temp_dir? I have .thumb files there since March, I don't think they are needed anymore but I'll investigate... 


Regarding the dependencies, there is nothing to remove:
image

But there is now the `ldap` php extension required (roundcube worked even without it, but composer requires it so better to install), I've added `apt install -y php7.2-ldap` at the end

**Comment by jvolkenant** [comment]
> Have either of you checked if Roundcube 1.4 has different dependencies that we should install (or any packages that are no longer needed)? The Roundcube upgrade notes also say "4. When upgrading from version older than 1.4.0 make sure old files in configured temp_dir are removed. Since this version we use constant filename prefix and do not remove files not starting with "RCMTEMP"." Do we need to do anything for that?

Diffing the INSTALL files from 1.3.10 and 1.4.0, it looks like there is no dependancy changes.

Post-Upgrade activites are the same between versions with the addtion of your mentioned temp_dir cleanup.  We define `$config['temp_dir'] = '/var/tmp/roundcubemail/';` in config.inc.php, that might need to get emptied if file names have changed.

**Comment by jvolkenant** [comment]
> 
> But there is now the `ldap` php extension required (roundcube worked even without it, but composer requires it so better to install), I've added `apt install -y php7.2-ldap` at the end

Please add php7.2-ldap to apt_install at the top of the script instead.

**Comment by fmonts** [comment]
Please check my latest commit, I check if the current version is 1.3, and only in that case run the delete command

**Comment by terry90** [comment]
Just for your info, I'm running roundcube with this patch, I ran setup/start.sh after rebasing on your PR and it's running well.
Maybe one thing I noticed is that my menu is frozen during filter creation:
`domain/?_task=settings&_action=plugin.managesieve` then click on create.
Everything else is OK for me.

**Comment by jvolkenant** [comment]
I was able to create a sieve rule, but not delete one.  I switched skins and the problem persists.  I believe it is related to this https://github.com/roundcube/roundcubemail/issues/7007

**Comment by fmonts** [comment]
Yes I confirm. It's already been fixed, the fix will be released in 1.4.1

**Comment by fmonts** [comment]
Or we could overwrite the file with the patched version, by adding a

`wget https://raw.githubusercontent.com/roundcube/roundcubemail/1ef912555d1d89df6fd580d946484797815faf96/plugins/managesieve/managesieve.js -O /usr/local/lib/roundcubemail/plugins/managesieve/managesieve.min.js`

**Comment by jvolkenant** [comment]
> 
> 
> Or we could overwrite the file with the patched version, by adding a
> 
> ```
> if [ $VERSION == 1.4.0 ]; then
>     # Patch managesieve.js. To be removed at the next update
>     wget https://raw.githubusercontent.com/roundcube/roundcubemail/1ef912555d1d89df6fd580d946484797815faf96/plugins/managesieve/managesieve.js -O /usr/local/lib/roundcubemail/plugins/managesieve/managesieve.min.js
> fi
> ```



> 
> 
> Or we could overwrite the file with the patched version, by adding a
> 
> ```
> if [ $VERSION == 1.4.0 ]; then
>     # Patch managesieve.js. To be removed at the next update
>     wget https://raw.githubusercontent.com/roundcube/roundcubemail/1ef912555d1d89df6fd580d946484797815faf96/plugins/managesieve/managesieve.js -O /usr/local/lib/roundcubemail/plugins/managesieve/managesieve.min.js
> fi
> ```

Roundcube has 29 commit's in 6 days.  Might be good to wait until 1.4.1.

**Comment by fmonts** [comment]
Roundcube 1.4.1 has just been released: https://github.com/roundcube/roundcubemail/releases/tag/1.4.1

I've updated the script.

**Comment by JoshData** [comment]
Thanks. The CHANGLOG needs to be updated with the new version. And according to the Roundcube release notes, we should also be able to delete

```
\$config['smtp_port'] = 587;
\$config['smtp_user'] = '%u';
\$config['smtp_pass'] = '%p';
```

because these are now the default values.

**Comment by fmonts** [comment]
> Thanks. The CHANGLOG needs to be updated with the new version. And according to the Roundcube release notes, we should also be able to delete
> 
> ```
> \$config['smtp_port'] = 587;
> \$config['smtp_user'] = '%u';
> \$config['smtp_pass'] = '%p';
> ```
> 
> because these are now the default values.

Done

**Comment by yodax** [comment]
This works for me. Email filters work too. Awesome work.

File: setup/webmail.sh [hunk 1]
```diff
@@ -51,6 +51,11 @@ elif [[ "$UPDATE_KEY" != `cat /usr/local/lib/roundcubemail/version` ]]; then
 	needs_update=1 #NODOC
 fi
 if [ $needs_update == 1 ]; then
+  # if upgrading from 1.3.x, clear the temp_dir
+  if [ "$(cat /usr/local/lib/roundcubemail/version | cut -c1-3)" == '1.3' ]; then
+    find /var/tmp/roundcubemail/ -type f ! -name 'RCMTEMP*' -delete
+  fi
+
```
[the remark this answers is not in the document]
Comment (jvolkenant) [reply setup/webmail.sh]: Thanks

**Pull request merged**